miniapps/con_gen/connection_generator.hpp

```diff
@@ -0,0 +1,261 @@
+
+            // Used to assure correct weight sign
+            float weight_sign = pro_pars.weight_mean < 0 ? -1 : 1;
+
+            // Check if this gid receives connections via this projection
+            // TODO: Replace with the fancy in range function we have somewhere in the utils
+            if (gid < post_pop.start_index || gid > (post_pop.start_index + post_pop.n_cells)) {
+                continue;
+            }
+
+            // Convert to the local gid of the post neuron
+            auto pop_local_gid = gid - post_pop.start_index;
+
+            // Convert this to a normalized location
+            std::pair<float, float> post_location = {
+                float(pop_local_gid % post_pop.x_side) / post_pop.x_side,
+                float(pop_local_gid / post_pop.y_side) / post_pop.y_side};
+
+            // If we have non square sides we need to correct the stdev to get
+            // circular projections!
+            float sd_x = pro_pars.sd;
+            float sd_y = pro_pars.sd;
+            if (post_pop.x_side != post_pop.y_side) {
+                if (post_pop.x_side < post_pop.y_side) {
+                    float ratio = float(post_pop.y_side) / post_pop.x_side;
+                    sd_x *= ratio;
+                }
+                else {
+                    float ratio = float(post_pop.x_side) / post_pop.y_side;
+                    sd_y *= ratio;
+                }
+            }
+
+            // Now we sample from the pre population based on the x,y location
+            // We supply the connection_impl with the size of the pre population
+            auto pre_locations = connection_locations(gen, post_location,
+                pro_pars.count, sd_x, sd_y, post_pop.periodic);
+
+            for (auto pre_location : pre_locations)
+            {
+                // convert the normalized locations to gid
+                cell_gid_type gid_pre = cell_gid_type(pre_location.second * pre_pop.y_side) * pre_pop.x_side +
+                    cell_gid_type(pre_location.first * pre_pop.x_side);
+
+                // Calculate the distance between the pre and post neuron.
+                float distance = std::sqrt(std::pow(pre_location.first * post_location.first, 2) +
+                    std::pow(pre_location.second * post_location.second, 2));
+
+                float delay = distance / pro_pars.delay_per_sd + pro_pars.delay_min;
+
+                float weight = weight_distr(gen);
+                // Flip the sign of the weight depending if we are incorrect
+                weight = (weight_sign * weight) < 0?  -weight: weight;
+
+                connections.push_back({ gid_pre + pre_pop.start_index, weight,  delay });
+            }
+        }
+
+        return connections;
+    }
+
+    std::vector<std::pair<float, float>> connection_locations(std::mt19937 gen,
+        std::pair<float, float> const target_location, unsigned count,
```

Comment:
The const here isn't really adding anything. The parameter is being passed by value.

Reply:
ok